src/m365/spo/commands/listitem/listitem-roleassignment-add.spec.ts

```diff
@@ -0,0 +1,421 @@
+        listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF',
+        listItemId: 1,
+        upn: 'someaccount@tenant.onmicrosoft.com',
+        roleDefinitionId: 1073741827
+      }
+    });
+  });
+
+  it('add role assignment to listitem in list get principal id by group name', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists(guid\'0CD891EF-AFCE-4E55-B836-FCE03286CCCF\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    sinon.stub(Cli, 'executeCommandWithOutput').callsFake((command): Promise<any> => {
+      if (command === SpoGroupGetCommand) {
+        return Promise.resolve({
+          stdout: '{"Id": 11,"IsHiddenInUI": false,"LoginName": "otherGroup","Title": "otherGroup","PrincipalType": 8,"AllowMembersEditMembership": false,"AllowRequestToJoinLeave": false,"AutoAcceptRequestToJoinLeave": false,"Description": "","OnlyAllowMembersViewMembership": true,"OwnerTitle": "Some Account","RequestToJoinLeaveEmailSetting": null}'
+        });
+      }
+
+      return Promise.reject(new CommandError('Unknown case'));
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF',
+        listItemId: 1,
+        groupName: 'someGroup',
+        roleDefinitionId: 1073741827
+      }
+    });
+  });
+
+  it('correctly handles error when group does not exist', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists(guid\'0CD891EF-AFCE-4E55-B836-FCE03286CCCF\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    const error = 'no group found';
+    sinon.stub(Cli, 'executeCommandWithOutput').callsFake((command): Promise<any> => {
+      if (command === SpoGroupGetCommand) {
+        return Promise.reject(error);
+      }
+
+      return Promise.reject(new CommandError('Unknown case'));
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF',
+        listItemId: 1,
+        groupName: 'someGroup',
+        roleDefinitionId: 1073741827
+      }
+    });
+  });
+
+  it('add role assignment to listitem in list get role definition id by role definition name', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists(guid\'0CD891EF-AFCE-4E55-B836-FCE03286CCCF\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
```

Comment:
Let's try to avoid the usage of `indexOf` and in place use a full comparison.
`if (opts.url === 'exampleurl')`